### PR TITLE
Fix/remote fork off

### PR DESCRIPTION
**Pull request by fbielejec**
Refactors the tool to use safe RPC methods (API stays the same)

### CODE REVIEW

File: fork-off/src/main.rs [hunk 1]
```diff
@@ -85,28 +84,104 @@ async fn main() -> anyhow::Result<()> {
...
         .await
         .expect("Could not deserialize response as JSON");
 
-    storage["result"]
+    debug!("get_key response: {}", response);
+
+    let result = response["result"]
         .as_array()
         .expect("No result in response")
+        .to_owned();
```
Comment (artemis-cc) [inline fork-off/src/main.rs:120]: We can save an allocation by dropping this `to_owned()`.

File: fork-off/src/main.rs [hunk 2]
```diff
@@ -85,28 +84,104 @@ async fn main() -> anyhow::Result<()> {
...
+                first_key = get_key(http_rpc_endpoint, &prefix, Some(&key)).await;
+            }
+
+            pairs
+        })
+        .collect::<Vec<Vec<(String, String)>>>()
+        .await
+        .into_iter()
+        .flatten()
+        .collect()
```
Comment (artemis-cc) [inline fork-off/src/main.rs:182]:
```suggestion
            futures::stream::iter(pairs)
        })
        .flatten()
        .collect::<Vec<(String, String)>>()
        .await
```

saves a collect-intoiter round trip.

File: fork-off/src/main.rs [hunk 3]
```diff
@@ -37,44 +44,36 @@ async fn main() -> anyhow::Result<()> {
...
-    let storage = get_chain_state(http_rpc_endpoint).await;
+    let hashed_prefixes = prefixes
+        .iter()
+        .map(|prefix| {
+            let hash = format!("0x{}", prefix_as_hex(prefix));
+            info!("prefix: {}, hash: {}", prefix, hash);
+            hash
+        })
+        .chain(vec![
+            "0x3a636f6465".to_string(), // code
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply fork-off/src/main.rs:65]: done

File: fork-off/src/main.rs [hunk 4]
```diff
@@ -37,44 +44,36 @@ async fn main() -> anyhow::Result<()> {
...
 
-    let storage = get_chain_state(http_rpc_endpoint).await;
+    let hashed_prefixes = prefixes
+        .iter()
+        .map(|prefix| {
+            let hash = format!("0x{}", prefix_as_hex(prefix));
+            info!("prefix: {}, hash: {}", prefix, hash);
+            hash
+        })
+        .chain(vec![
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply fork-off/src/main.rs:64]: fixed

File: fork-off/src/main.rs (hunk 1 above)
Comment (fbielejec) [reply fork-off/src/main.rs]: done

**Review by kostekIV — APPROVED** [review]
lgtm

**Review by bartoszjedrzejewski — APPROVED** [review]
Tested against testnet env, seems to work and create a new chainspec file. LGTM.

**Pull request merged**